hibernate-core/src/main/java/org/hibernate/query/sql/internal/ParameterParser.java

```diff
@@ -178,6 +176,14 @@ else if ( c == '?' ) {
 		recognizer.complete();
 	}
 
+	private static String preprocessing(String sqlString) {
+		final Map<String, String> preprocessingExchangeMap = Map.of("::=", ":=");
```

Comment:
I guess if we want to stay backwards compatible, then we should also handle `::::` specially as that syntax seems to have been the workaround so far.

```suggestion
		final Map<String, String> preprocessingExchangeMap = Map.of( "::=", ":=", "::::", "::" );
```